superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.cpp

```diff
@@ -0,0 +1,507 @@
+                                                         IID_PPV_ARGS(dmlOperatorInitializer.GetAddressOf())));
+
+    // Query the operator for the required size (in descriptors) of its binding table.
+    // You need to initialize an operator exactly once before it can be executed, and
+    // the two stages require different numbers of descriptors for binding. For simplicity,
+    // we create a single descriptor heap that's large enough to satisfy them both.
+    DML_BINDING_PROPERTIES initializeBindingProperties = dmlOperatorInitializer->GetBindingProperties();
+    DML_BINDING_PROPERTIES executeBindingProperties = m_dmlCompiledOperator->GetBindingProperties();
+    UINT descriptorCount =
+        initializeBindingProperties.RequiredDescriptorCount > executeBindingProperties.RequiredDescriptorCount
+            ? initializeBindingProperties.RequiredDescriptorCount
+            : executeBindingProperties.RequiredDescriptorCount;
+
+    // Create descriptor heaps.
+    ComPtr<ID3D12DescriptorHeap> descriptorHeap;
+    D3D12_DESCRIPTOR_HEAP_DESC descriptorHeapDesc{};
+    descriptorHeapDesc.Type = D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV;
+    descriptorHeapDesc.NumDescriptors = descriptorCount;
+    descriptorHeapDesc.Flags = D3D12_DESCRIPTOR_HEAP_FLAG_SHADER_VISIBLE;
+    ThrowIfFailed(m_device->CreateDescriptorHeap(&descriptorHeapDesc, _uuidof(descriptorHeap),
+                                                 (void **)descriptorHeap.GetAddressOf()));
+
+    // Set the descriptor heap(s).
+    ID3D12DescriptorHeap *d3D12DescriptorHeaps[] = {descriptorHeap.Get()};
+    m_commandList->SetDescriptorHeaps(ARRAYSIZE(d3D12DescriptorHeaps), d3D12DescriptorHeaps);
+
+    // Create a binding table over the descriptor heap we just created
+    DML_BINDING_TABLE_DESC dmlBindingTableDesc{};
+    dmlBindingTableDesc.Dispatchable = dmlOperatorInitializer.Get();
+    dmlBindingTableDesc.CPUDescriptorHandle = descriptorHeap->GetCPUDescriptorHandleForHeapStart();
+    dmlBindingTableDesc.GPUDescriptorHandle = descriptorHeap->GetGPUDescriptorHandleForHeapStart();
+    dmlBindingTableDesc.SizeInDescriptors = descriptorCount;
+    ThrowIfFailed(m_dmlDevice->CreateBindingTable(&dmlBindingTableDesc, IID_PPV_ARGS(m_bindingTable.GetAddressOf())));
+
+    // Create the temporary and persistent resources that are necessary for executing an operator.
+    // The temporary resource is scratch memory (used internally by DirectML), whose contents you don't need to define.
+    // The persistent resource is long-lived, and you need to initialize it using the IDMLOperatorInitializer.
+    UINT64 temporaryResourceSize =
+        max(initializeBindingProperties.TemporaryResourceSize, executeBindingProperties.TemporaryResourceSize);
+    UINT64 persistentResourceSize = executeBindingProperties.PersistentResourceSize;
+
+    // Bind and initialize the operator on the GPU.
+    ComPtr<ID3D12Resource> temporaryBuffer;
+    if (temporaryResourceSize != 0) {
```

Comment:
>  if (temporaryResourceSize != 0) { [](http://example.com/codeflow?start=2&length=35)

what happens if this value is 0? Can we continue processing or we need stop the execution?

Reply:
just continue processing, it's a template code from the tutorial, in our case temporaryResourceSize is truly 0